play.pokemonshowdown.com/src/panels.tsx

```diff
@@ -349,6 +349,37 @@ export class PSView extends preact.Component {
 			return PS.prefs.refreshprompt ? "Are you sure you want to leave?" : null;
 		};
 
+		window.addEventListener('submit', ev => {
+			let elem = ev.target as HTMLFormElement | null;
+			if (elem?.getAttribute('data-submitsend')) {
+				const elements = elem.querySelectorAll('input[name], select[name], textarea[name], keygen[name], button[value]');
+				const inputs = [];
+				for (const element of Array.from(elements)) {
+					const type = element.getAttribute('type') || "";
+					const checked = element.getAttribute('checked');
```

Comment:
!!! Wait, this is wrong. The attribute is the actual HTML source, which affects whether the checkbox _starts_ checked, not whether it's currently checked.

```suggestion
					const checked = (element as HTMLInputElement).checked;
```